freqtrade/exchange/exchange.py

```diff
@@ -1525,6 +1581,61 @@ def get_historic_trades(self, pair: str,
             self._async_get_trade_history(pair=pair, since=since,
                                           until=until, from_id=from_id))
 
+    @retrier
+    def fill_leverage_brackets(self):
+        """
+            #TODO-lev: Should maybe be renamed, leverage_brackets might not be accurate for kraken
+            Assigns property _leverage_brackets to a dictionary of information about the leverage
+            allowed on each pair
+        """
+        raise OperationalException(
+            f"{self.name.capitalize()}.fill_leverage_brackets has not been implemented.")
+
+    def get_max_leverage(self, pair: Optional[str], nominal_value: Optional[float]) -> float:
+        """
+            Returns the maximum leverage that a pair can be traded at
+            :param pair: The base/quote currency pair being traded
+            :nominal_value: The total value of the trade in quote currency (collateral + debt)
+        """
+        raise OperationalException(
+            f"{self.name.capitalize()}.get_max_leverage has not been implemented.")
+
+    @retrier
+    def set_leverage(self, leverage: float, pair: Optional[str]):
+        """
+            Set's the leverage before making a trade, in order to not
+            have the same leverage on every trade
+        """
+        try:
+            self._api.set_leverage(symbol=pair, leverage=leverage)
```

Comment:
I'm pretty sceptical on having this in `exchange.py`.

Checking with an uptodate ccxt version, this seems only supported for binance and ftx ... so at the very least, it needs a check if it's implemented  (using `self.exchange_has('setLeverage')`

Having that check will also make the "fake method" in the kraken class obsolete...

Reply:
This is the unified method, so if other exchanges don't have it, it's because ccxt hasn't implemented it. I don't like having it in the subclasses because it's going to just be the same code duplicated. If it's in the exchange class, then once ccxt implements it for exchanges that we don't even have in freqtrade, it will still work for those exchanges, but if it's in the subclasses then it won't and we would have to implement it for every exchange that we want freqtrade to be able to use with leverage.

I'll put the check in